src/Core/DynamicMetadata/DynamicMetadataInterface.php

```diff
@@ -10,7 +10,7 @@
     /**
      * @return string The name of the metadata
      */
-    public function getName();
+    public static function getName();
```

Comment:
this feels odd. why would we need a static call to find out the name? if we use the dynamic metadata statically, we don't need an interface.